### PR TITLE
Geometry Management for Google Maps

### PR DESCRIPTION
**Pull request by alonsogarciapablo**
Fixes https://github.com/CartoDB/cartodb.js/issues/1519.

Implements geometry management (drawing + edition for Google Maps).

Here's the new class hierarchy for geometry views implemented [here](https://github.com/CartoDB/cartodb.js/tree/4d6b5b20798d5de8a1668e5e532ff545312f23eb/src/geo/geometry-views):

```
geometry-view-base.js
├── point-view-base.js
│   ├── leaflet-point-view.js -> markerAdapter
│   ├── gmaps-point-view.js -> markerAdapter
├── path-view-base.js
│   ├── leaflet-polyline-view.js -> pathAdapter
│   ├── leaflet-polygon-view.js -> pathAdapter 
│   ├── gmaps-polyline-view.js -> pathAdapter
│   ├── gmaps-polygon-view.js -> pathAdapter
├── multi-geometry-view-base.js
│   ├── leaflet-multi-point-view.js -> geometryView
│   ├── leaflet-multi-polyline-view.js -> geometryView
│   ├── leaflet-multi-polygon-view.js -> geometryView
│   ├── gmaps-multi-point-view.js -> geometryView
│   ├── gmaps-multi-polyline-view.js -> geometryView
│   ├── gmaps-multi-polygon-view.js -> geometryView
├── marker-adapter-base.js
│   ├── leaflet-marker-adapter.js -> L.marker
│   ├── gmaps-marker-adapter.js -> google.maps.Marker
├── path-adapter-base.js
    ├── leaflet-path-adapter.js -> L.polyline | L.polygon
    ├── gmaps-path-adapter.js -> google.maps.Polyline | google.maps.Polygon
```

**Acceptance:**

- Checkout `google-maps-geometry-management`.
- Run `grunt dev`.
- Make sure the following examples are working correctly (draw geometries, edit existing ones, etc.):
  - http://localhost:9001/examples/v4/leaflet-geometry-editor.html
  - http://localhost:9001/examples/v4/gmaps-geometry-editor.html

cc: @rochoa

### CODE REVIEW

File: themes/scss/infowindow/cartodb-infowindow-default.scss [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 .CDB-infowindow-wrapper {
   position: absolute;
+  z-index: 1; // makes infowindows visible with Google Maps
```
Comment (alonsogarciapablo) [inline themes/scss/infowindow/cartodb-infowindow-default.scss]: cc: @piensaenpixel ☝️

**Comment by alonsogarciapablo** [comment]
Retest this please

**Comment by alonsogarciapablo** [comment]
Retest this please

**Comment by rochoa** [comment]
I just tested it and both, gmaps and leaflet, have the same behavior.

The only thing that it is still weird for me is not having _midpoints_ when creating a new polygon/polyline, but I guess that's the expected behavior.

**Comment by alonsogarciapablo** [comment]
Thanks for the CR and for testing this @rochoa!

> The only thing that it is still weird for me is not having midpoints when creating a new polygon/polyline, but I guess that's the expected behavior.

Yes, this is something we decided early on.

**Pull request merged**